sdk/identity/azure-identity/inc/azure/identity/default_azure_credential.hpp

```diff
@@ -25,10 +25,10 @@ namespace Azure { namespace Identity {
    * as well.
    *
    * @details This credential is using several credentials in the following order:
-   * #Azure::Identity::EnvironmentCredential, #Azure::Identity::AzureCliCredential, and
-   * #Azure::Identity::ManagedIdentityCredential. Even though the credentials being used and their
-   * order is documented, it may be changed in the future versions of the SDK, potentially bringing
-   * breaking changes in its behavior.
+   * #Azure::Identity::EnvironmentCredential, #Azure::Identity::WorkloadIdentityCredential,
+   * #Azure::Identity::AzureCliCredential, and #Azure::Identity::ManagedIdentityCredential. Even
+   * though the credentials being used and their order is documented, it may be changed in the
```

Comment:
From my reading of it (and some background context), it is the opposite, where we want to make it clear through docs that we wouldn't consider this a breaking change, and reserve the right to adjust it.

For example, here's the snippet that conveys the intent clearly:
> The good thing is that everywhere in documentation we say that although we are documenting the order of credentials, we may change their order in the future, and that would not be considered a breaking change.

We intend to update the order to match other SDKs shortly. I can make a note of that in the "Other Changes" within the Changelog at that time:
https://github.com/Azure/azure-sdk-for-cpp/issues/4669

Reply:
While this is certainly a disclaimer that it might break in the future, I believe that regardless of that disclaimer, it's our responsibility to announce any changes that might be breaking. 

And adding a new auth provider or changing the order of auth providers is potentially a breaking change.